index.d.ts

```diff
@@ -526,6 +526,24 @@ declare module 'pagarme' {
     birthday?: string;
   }
 
+  export interface CustomerBilletInput {
```

Comment:
@mrlmaia  Já temos uma _interface_ para _input_ do _customer_ , chamada `CustomerInput`. Qual  a diferença entre `CustomerBilletInput` e `CustomerInput`.  Se forem idênticas não há necessidade de criar outra

Reply:
Os campos são os mesmos, a diferença é que alguns campos como email e phone_numbers são opcionais

Reply:
Entendi. Então nesse caso talvez fosse interessante manter uma interface de base para o CustomerInput só que ao invés de copiar a interface inteira e colar, VC pode criar uma nova type chamada CustomerInputBoleto ou CustomerInputCreditCard usando a type utilitária Required ou Partial.
Ao meu ver, o CustomerInput deveria ser a BASE, ou seja, confira o método de pagamento to que muda menos o CustomerInput e use ele para essa nova type.
Suponhamos que no método de pagamento boleto, o nome é obrigatório e essa é a única diferença entre eles. Então a nossa base vai ser a do cliente no formato cartão de credito e criaremos uma nova type que iria copiar CustomerInput usando Requires e dar o nome de CustomerInputBoleto. Dessa forma teremos dois types. Então é usar eles juntos para montar a interface da transaction